### PR TITLE
Implement 'serve_savedmodel()' to serve models

### PR DESCRIPTION
**Pull request by javierluraschi**
Adds support for `serve_savedmodel()` as a helper function to locally deploy TensorFlow saved models. Usage:

```r
library(tensorflow)
sess <- tf$Session()

# (1) Train MNIST model with 'x' as input and 'y' as output.

# (2) Save model with signature:

model_dir <- "trained"
builder <- tf$saved_model$builder$SavedModelBuilder(model_dir)
builder$add_meta_graph_and_variables(
  sess,
  list(
    tf$python$saved_model$tag_constants$SERVING
  ),
  signature_def_map = list(
    serving_default = tf$saved_model$signature_def_utils$build_signature_def(
      inputs = list(images = tf$saved_model$utils$build_tensor_info(x)),
      outputs = list(scores = tf$saved_model$utils$build_tensor_info(y))
    )
  )
)
builder$save()

# (3) Serve the model

serve_savedmodel(model_dir)
```

screen shot 2017-10-26 at 5 02 15 pm

### CODE REVIEW

File: R/serve.R [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      )
+    },
+    ".*" = function(req, signature_def) {
+      serve_invalid_request()
+    }
+  )
+}
+
+serve_run <- function(model_dir, host, port, start, browse) {
+  sess <- tf$Session()
```
Comment (jjallaire) [inline R/serve.R]: Probably . need `on.exit(ses$close(), add = TRUE)` here.

File: R/view.R [hunk 2]
```diff
@@ -19,6 +19,7 @@ view_savedmodel <- function(
   compat <- tf$python$util$compat
   saved_model_pb2 <- tf$core$protobuf$saved_model_pb2
 
+  sess <- NULL
   with(tf$Session() %as% sess, {
```
Comment (jjallaire) [inline R/view.R]: Might want to implement a custom `with` wrapper than closes the session when exiting the scope. You could also support using any default session that exists. For example: https://github.com/rstudio/tfdatasets/blob/master/R/utils.R#L24-L32

File: R/swagger.R [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      ),
+      produces = list(
+        unbox("application/json")
+      ),
+      parameters = list(
+        list(
+          "in" = unbox("body"),
+          name = unbox("body"),
+          description = unbox(paste0("Prediction instances for '", siagnature_name, "'")),
+          required = unbox(TRUE),
```
Comment (terrytangyuan) [inline R/swagger.R]: siagnature_name > signature_name

File: R/serve.R (hunk 1 above)
Comment (javierluraschi) [reply R/serve.R]: Right! Changed.

File: R/swagger.R (hunk 3 above)
Comment (javierluraschi) [reply R/swagger.R]: Thanks, changed.

File: R/view.R (hunk 2 above)
Comment (javierluraschi) [reply R/view.R]: Added `with_new_session`.

**Comment by javierluraschi** [comment]
Closing to move `serve_savedmodel()` into a new package.

**Pull request closed without merging**